jedi/evaluate/pep0484.py

```diff
@@ -80,9 +80,66 @@ def _fix_forward_reference(context, node):
         return node
 
 
+def _split_mypy_param_declaration(decl_text):
+    """
+    Split decl_text on commas, but group generic expressions
+    together.
+
+    For example, given "foo, Bar[baz, biz]" we return
+    ['foo', 'Bar[baz, biz]'].
+
+    """
+    node = parse(decl_text).children[0]
```

Comment:
Here we need to handle the error cases. Please add a test with a comment that is invalid. You can either just catch the parser error and return `[]` or use error_recovery=True and then handle all the issues.